app/controllers/sections/info/DispatchDetailsController.scala

```diff
@@ -43,38 +43,92 @@ class DispatchDetailsController @Inject()(
                                            val getData: DataRetrievalAction,
                                            val requireData: DataRequiredAction,
                                            formProvider: DispatchDetailsFormProvider,
+                                           val userAnswersService: UserAnswersService,
                                            val controllerComponents: MessagesControllerComponents,
                                            view: DispatchDetailsView,
                                            val userAllowList: UserAllowListAction
                                          ) extends BasePreDraftNavigationController with AuthActionHelper with PreDraftAuthActionHelper {
 
   def onPreDraftPageLoad(ern: String, mode: Mode): Action[AnyContent] =
     authorisedPreDraftDataRequestAsync(ern) { implicit request =>
-      renderView(Ok, fillForm(DispatchDetailsPage, formProvider()), mode)
+      renderViewPreDraft(
+        Ok,
+        fillForm(DispatchDetailsPage(), formProvider()),
+        controllers.sections.info.routes.DispatchDetailsController.onPreDraftSubmit(request.ern, mode),
+        mode
+      )
     }
 
 
   def onPreDraftSubmit(ern: String, mode: Mode): Action[AnyContent] =
     authorisedPreDraftDataRequestAsync(ern) { implicit request =>
       formProvider().bindFromRequest().fold(
         formWithErrors =>
-          renderView(BadRequest, formWithErrors, mode),
+          renderViewPreDraft(
+            BadRequest,
+            formWithErrors,
+            controllers.sections.info.routes.DispatchDetailsController.onPreDraftSubmit(request.ern, mode),
+            mode
+          ),
+        value =>
+          savePreDraftAndRedirect(DispatchDetailsPage(), value, mode)
+      )
+    }
+
+  def onPageLoad(ern: String, draftId: String): Action[AnyContent] =
```

Comment:
do we need a mode?

Reply:
No, you're only ever going to hit this post-draft from the CYA page so you'll only ever be changing it (mode is inferred to always be CheckMode)